packages/app-builder-lib/src/asar/asar.ts

```diff
@@ -2,6 +2,20 @@ import { createFromBuffer } from "chromium-pickle-js"
 import { close, open, read, readFile, Stats } from "fs-extra"
 import * as path from "path"
 
+/** @internal */
+export interface ReadAsarHeader {
```

Comment:
@mmaietta I don't think this, below, and `AsarFilesystem` should be internal. I get following TS errors when importing `electron-builder`:

```
node_modules/app-builder-lib/out/asar/asar.d.ts:1:66 - error TS2304: Cannot find name 'ReadAsarHeader'.

1 export declare function readAsarHeader(archive: string): Promise<ReadAsarHeader>;
                                                                   ~~~~~~~~~~~~~~

node_modules/app-builder-lib/out/asar/asar.d.ts:2:60 - error TS2304: Cannot find name 'AsarFilesystem'.

2 export declare function readAsar(archive: string): Promise<AsarFilesystem>;
                                                             ~~~~~~~~~~~~~~

node_modules/app-builder-lib/out/asar/integrity.d.ts:2:10 - error TS2305: Module '"./asar"' has no exported member 'NodeIntegrity'.

2 import { NodeIntegrity } from "./asar";
           ~~~~~~~~~~~~~

[10:10:15 AM] Found 3 errors. Watching for file changes.
```

Reply:
I have no qualms with making them public. Can you open a PR for it?

Reply:
Sure thing. Done: https://github.com/electron-userland/electron-builder/pull/6692 Thank you!

Reply:
btw, I'm happy to report that we have tested 23.0.1 on our app and both asar integrity and merging ASARs appears to be working great! Thanks!